gateway/exporters/statsd/statsd.go

```diff
@@ -141,38 +146,43 @@ func (e *StatsdExporter) Export(leaf *ctree.Leaf) {
 		}
 
 		metric.Dims = point.Tags
-		metric.Account = metric.Dims["Account"]
-		delete(metric.Dims, "Account")
 		// ns since epoch
 		metric.Dims["timestamp"] = strconv.FormatInt(notification.Timestamp, 10)
 
+		notificationType := e.config.GetPathMetadata(path)["type"]
+		e.config.Log.Debug().Msgf("Notification type: [ %s ]", notificationType)
+		if notificationType == "" || notificationType == MetricType || notificationType == LoggedMetricType {
+			metric.Account = e.config.Exporters.GenevaMdmAccount
+			if e.config.Exporters.GenevaMdmAccount == "" {
+				e.config.Log.Warn().Msg("geneva MDM account is not set in exporter; metrics will be discarded by the MDM agent")
+			}
 
-		metricJSON, err := json.Marshal(metric)
+			if resourceId := e.config.Exporters.ExtensionArmId; resourceId != "" {
+				metric.Dims["microsoft.resourceid"] = resourceId
+			} else {
+				e.config.Log.Warn().Msg("extension ARM ID is not set in Geneva exporter; metrics will be discarded by the MDM agent")
```

Comment:
this warning should be logged when the exporter is initialized, not for every metric. there could be thousands of them per second.